tests/util_bucket.sh

```diff
@@ -0,0 +1,304 @@
+delete_bucket_recursive_s3api() {
+  log 6 "delete_bucket_recursive_s3api"
+  if [ $# -ne 1 ]; then
+    log 2 "'delete_bucket_recursive_s3api' requires bucket name"
+    return 1
+  fi
+
+  if ! clear_bucket_s3api "$1"; then
+    log 2 "error clearing bucket (s3api)"
+    return 1
+  fi
+
+  if ! delete_bucket 's3api' "$1"; then
+    log 2 "error deleting bucket"
+    return 1
+  fi
+  return 0
+}
+
+# params: client, bucket name
+# return 0 on success, 1 on error
+delete_bucket_contents() {
+  log 6 "delete_bucket_contents"
+  if [ $# -ne 2 ]; then
+    log 2 "'delete_bucket_contents' requires client, bucket name"
+    return 1
+  fi
+
+  local exit_code=0
+  local error
+  if [[ $1 == "aws" ]] || [[ $1 == 's3api' ]]; then
+    if ! clear_bucket_s3api "$2"; then
+      log 2 "error clearing bucket (s3api)"
+      return 1
+    fi
+  elif [[ $1 == "s3cmd" ]]; then
+    delete_bucket_recursive "s3cmd" "$1"
+  elif [[ $1 == "mc" ]]; then
+    delete_bucket_recursive "mc" "$1"
+  elif [[ $1 == "s3" ]]; then
+    delete_bucket_recursive "s3" "$1"
+  else
+    log 2 "unrecognized client: '$1'"
+    return 1
+  fi
+  return 0
+}
+
+# check if bucket exists
+# param:  bucket name
+# return 0 for true, 1 for false, 2 for error
+bucket_exists() {
+  if [ $# -ne 2 ]; then
+    log 2 "bucket_exists command requires client, bucket name"
+    return 2
+  fi
+  local exists=0
+  head_bucket "$1" "$2" || exists=$?
+  # shellcheck disable=SC2181
+  if [ $exists -ne 0 ] && [ $exists -ne 1 ]; then
+    log 2 "unexpected error checking if bucket exists"
+    return 2
+  fi
+  if [ $exists -eq 0 ]; then
+    return 0
+  fi
+  return 1
+}
+
+# params:  client, bucket name
+# return 0 for success, 1 for error
+delete_bucket_or_contents() {
+  log 6 "delete_bucket_or_contents"
+  if [ $# -ne 2 ]; then
+    log 2 "'delete_bucket_or_contents' requires client, bucket name"
+    return 1
+  fi
+  if [[ $RECREATE_BUCKETS == "false" ]]; then
+    if ! delete_bucket_contents "$1" "$2"; then
+      log 2 "error deleting bucket contents"
+      return 1
+    fi
+
+    if ! delete_bucket_policy "$1" "$2"; then
+      log 2 "error deleting bucket policy"
+      return 1
+    fi
+
+    if ! get_object_ownership_rule_and_update_acl "$2"; then
+      log 2 "error getting object ownership rule and updating ACL"
+      return 1
+    fi
+
+    if ! abort_all_multipart_uploads "$2"; then
+      log 2 "error aborting all multipart uploads"
+      return 1
+    fi
+
+    if [ "$RUN_USERS" == "true" ] && ! reset_bucket_owner "$2"; then
+      log 2 "error resetting bucket owner"
+      return 1
+    fi
+
+    log 5 "bucket contents, policy, ACL deletion success"
+    return 0
+  fi
+  if ! delete_bucket_recursive "$1" "$2"; then
+    log 2 "error with recursive bucket delete"
+    return 1
+  fi
+  log 5 "bucket deletion success"
+  return 0
+}
+
+# params: client, bucket name
+# return 0 for success, 1 for error
+delete_bucket_or_contents_if_exists() {
```

Comment:
Same question here?  Seems like the 'or' clause belongs a level higher in the code and not in a method name.